fe/fe-core/src/main/java/com/starrocks/qe/DefaultCoordinator.java

```diff
@@ -882,22 +883,39 @@ public void updateFragmentExecStatus(TReportExecStatusParams params) {
     }
 
     @Override
-    public void updateAuditStatistics(TReportAuditStatisticsParams params) {
-        auditStatistics = new PQueryStatistics();
-        auditStatistics.scanRows = params.scan_rows;
-        auditStatistics.scanBytes = params.scan_bytes;
-        auditStatistics.returnedRows = params.returned_rows;
-        auditStatistics.cpuCostNs = params.cpu_cost_ns;
-        auditStatistics.memCostBytes = params.mem_cost_bytes;
-        auditStatistics.spillBytes = params.spill_bytes;
-        if (CollectionUtils.isNotEmpty(params.stats_items)) {
+    public synchronized void updateAuditStatistics(TReportAuditStatisticsParams params) {
```

Comment:
data sink is in the one single node, and you already have this check  `    if (_num_sinkers.fetch_sub(1, std::memory_order_acq_rel) == 1) {`, why this method will be concurrent?

Reply:
I'm not sure about that, maybe there are many backend will call this function concurrently.